hw/ip/rstmgr/data/rstmgr.sv.tpl

```diff
@@ -160,31 +161,51 @@ module rstmgr import rstmgr_pkg::*; (
   // Reset info construction                        //
   ////////////////////////////////////////////////////
 
-  logic [ResetReasons-1:0] rst_reqs;
   logic rst_hw_req;
   logic rst_low_power;
+  logic rst_ndm;
+  logic rst_cpu_nq;
+  logic first_reset;
 
-  assign rst_hw_req = pwr_i.reset_cause == pwrmgr_pkg::HwReq;
-  assign rst_low_power = pwr_i.reset_cause == pwrmgr_pkg::LowPwrEntry;
-
-  assign rst_reqs = {
-                    ndm_req_valid,
-                    rst_hw_req ? peri_i.rst_reqs : ExtResetReasons'(0),
-                    rst_low_power
-                    };
+  // The qualification of first reset below could technically be POR as well.
+  // However, that would enforce software to clear POR upon cold power up.  While that is
+  // the most likely outcome anyways, hardware should not require that.
+  assign rst_hw_req    = ~first_reset & pwr_i.reset_cause == pwrmgr_pkg::HwReq;
+  assign rst_ndm       = ~first_reset & ndm_req_valid;
+  assign rst_low_power = ~first_reset & pwr_i.reset_cause == pwrmgr_pkg::LowPwrEntry;
 
-  rstmgr_info #(
-    .Reasons(ResetReasons)
-  ) i_info (
+  prim_flop_2sync #(
+    .Width(1),
+    .ResetValue('0)
+  ) u_cpu_reset_synced (
     .clk_i,
-    .rst_ni(rst_por_aon_n),
-    .rst_cpu_ni(cpu_i.rst_cpu_n),
-    .rst_req_i(rst_reqs),
-    .wr_i(reg2hw.reset_info.qe),
-    .data_i(reg2hw.reset_info.q),
-    .rst_reasons_o(hw2reg.reset_info)
+    .rst_ni(local_rst_n),
+    .d_i(cpu_i.rst_cpu_n),
+    .q_o(rst_cpu_nq)
   );
 
+  // first reset is a flag that blocks reset recording until first de-assertion
+  always_ff @(posedge clk_i or negedge local_rst_n) begin
+    if (!local_rst_n) begin
+      first_reset <= 1'b1;
+    end else if (rst_cpu_nq) begin
+      first_reset <= 1'b0;
+    end
+  end
+
+  // Only sw is allowed to clear a reset reason, hw is only allowed to set it.
+  assign hw2reg.reset_info.low_power_exit.d  = 1'b1;
+  assign hw2reg.reset_info.low_power_exit.de = rst_low_power;
+
+  assign hw2reg.reset_info.ndm_reset.d  = 1'b1;
+  assign hw2reg.reset_info.ndm_reset.de = rst_ndm;
+
+  // HW reset requests most likely will be multi-bit, so OR in whatever reasons
+  // that are already set.
+  assign hw2reg.reset_info.hw_req.d  = pwr_i.rstreqs | reg2hw.reset_info.hw_req.q;
```

Comment:
do we want to report multiple reset requests, or only the one that actually came in first (and hence triggered the reset)?

Reply:
i plan on just reporting all of them.  Especially for the peripheral triggered ones, since the effect is identical. 
I discussed this awhile ago with the software guys and the thinking was that it should not matter too much.

Reply:
i should clarify..the reason is once the reset hits, the system is "wiped" out, so the reset requests should in theory all disappear.

Reply:
ah ok got it. makes sense, just wanted to clarify.